tiledb/sm/query/reader_base.h

```diff
@@ -346,50 +346,158 @@ class ReaderBase : public StrategyBase {
    * @return Status
    */
   Status unfilter_tile(
-      const std::string& name, Tile* tile, Tile* tile_var) const;
+      uint64_t num_range_threads,
+      uint64_t thread_idx,
+      const std::string& name,
+      Tile* tile,
+      const ChunkData& tile_chunk_data,
+      Tile* tile_var,
+      const ChunkData& tile_var_chunk_data) const;
 
   /**
    * Runs the input fixed-sized tile for the input nullable attribute
    * through the filter pipeline. The tile buffer is modified to contain the
    * output of the pipeline.
    *
+   * @param num_range_threads Total number of range threads.
+   * @param range_thread_idx Current range thread index.
    * @param name The attribute/dimension the tile belong to.
    * @param tile The tile to be unfiltered.
+   * @param tile_chunk_data The tile chunk info, buffers and offsets
    * @param tile_validity The validity tile to be unfiltered.
+   * @param tile_validity_chunk_data The validity tile chunk info, buffers and
+   * offsets
    * @return Status
    */
   Status unfilter_tile_nullable(
-      const std::string& name, Tile* tile, Tile* tile_validity) const;
+      uint64_t num_range_threads,
+      uint64_t thread_idx,
+      const std::string& name,
+      Tile* tile,
+      const ChunkData& tile_chunk_data,
+      Tile* tile_validity,
+      const ChunkData& tile_validity_chunk_data) const;
 
   /**
    * Runs the input var-sized tile for the input nullable attribute through
    * the filter pipeline. The tile buffer is modified to contain the output of
    * the pipeline.
    *
+   * @param num_range_threads Total number of range threads.
+   * @param range_thread_idx Current range thread index.
    * @param name The attribute/dimension the tile belong to.
    * @param tile The offsets tile to be unfiltered.
+   * @param tile_chunk_data The offsets tile chunk info, buffers and offsets
    * @param tile_var The value tile to be unfiltered.
+   * @param tile_var_chunk_data The value tile chunk info, buffers and offsets
    * @param tile_validity The validity tile to be unfiltered.
+   * @param tile_validity_chunk_data The validity tile chunk info, buffers and
+   * offsets
    * @return Status
    */
   Status unfilter_tile_nullable(
+      uint64_t num_range_threads,
+      uint64_t thread_idx,
+      const std::string& name,
+      Tile* tile,
+      const ChunkData& tile_chunk_data,
+      Tile* tile_var,
+      const ChunkData& tile_var_chunk_data,
+      Tile* tile_validity,
+      const ChunkData& tile_validity_chunk_data) const;
+
+  /**
+   * Filters the tiles on a particular attribute/dimension from all input
+   * fragments based on the tile info in `result_tiles`. Used only by legacy
+   * readers.
+   *
+   * @param name Attribute/dimension whose tiles will be unfiltered.
+   * @param result_tiles Vector containing the tiles to be unfiltered.
+   * @param disable_cache Disable the filtered buffers cache or not.
+   * @return Status
+   */
+  Status unfilter_tiles_legacy(
```

Comment:
We shouldn't use the word legacy here. For some scenarios, ie row major reads, what we've been calling the legacy reader is actually the only reader 🙃 Also, the refactored dense reader still uses the tile cache (for now) and will go down that path. Maybe we rename the unfilter_* APIs used by the refactored readers with _chunk_range and keep those with the original range?